packages/app/src/components/Admin/UserGroup/UserGroupDeleteModal.tsx

```diff
@@ -15,7 +17,33 @@ import AppContainer from '~/client/services/AppContainer';
  * @class GrantSelector
  * @extends {React.Component}
  */
-class UserGroupDeleteModal extends React.Component {
+type Props = {
+  t: TFunction, // i18next
+  appContainer: AppContainer,
+
+  userGroups: IUserGroup[],
+  deleteUserGroup?: IUserGroup,
+  onDelete?: (deleteGroupId: string, actionName: string, transferToUserGroupId: string) => Promise<void> | void,
+  isShow: boolean,
+  onShow?: (group: IUserGroup) => Promise<void> | void,
+  onHide?: () => Promise<void> | void,
+};
+type State = {
+  actionName: string,
+  transferToUserGroupId: string,
+};
+
+class UserGroupDeleteModal extends React.Component<Props, State> {
+
+  actionForPages: any;
+
+  availableOptions: any;
+
+  xss: Xss;
+
+  state: State;
```

Comment:
https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/class_components
ここに型をもう一回宣言しておくと若干いいことあると書いてあったのでつけました
```
It isn't strictly necessary to annotate the state class property, but it allows better type inference when accessing this.state and also initializing the state.

This is because they work in two different ways, the 2nd generic type parameter will allow this.setState() to work correctly, because that method comes from the base class, but initializing state inside the component overrides the base implementation so you have to make sure that you tell the compiler that you're not actually doing anything different.
```